tests/integration/cairo1/out.json

```diff
@@ -0,0 +1,194 @@
+    "0x9",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127ff47fff8000",
+    "0x48127ff47fff8000",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x1",
+    "0x480680017fff8000",
+    "0x2",
+    "0x1104800180018000",
+    "0x52",
+    "0x20680017fff7ffd",
+    "0x6",
+    "0x10780017fff7fff",
+    "0x2",
+    "0x10780017fff7fff",
+    "0x7",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127ffd7fff8000",
+    "0x48127ffd7fff8000",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x1",
+    "0x480680017fff8000",
+    "0x2",
+    "0x1104800180018000",
+    "0x3a",
+    "0x20680017fff7ffd",
+    "0x6",
+    "0x10780017fff7fff",
+    "0x2",
+    "0x10780017fff7fff",
+    "0x9",
+    "0x40780017fff7fff",
+    "0x12",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127feb7fff8000",
+    "0x48127feb7fff8000",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x1",
+    "0x480680017fff8000",
+    "0x2",
+    "0x1104800180018000",
+    "0x27",
+    "0x20680017fff7ffd",
+    "0x6",
+    "0x10780017fff7fff",
+    "0x2",
+    "0x10780017fff7fff",
+    "0x9",
+    "0x40780017fff7fff",
+    "0x9",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127ff47fff8000",
+    "0x48127ff47fff8000",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x1",
+    "0x480680017fff8000",
+    "0x2",
+    "0x1104800180018000",
+    "0x14",
+    "0x20680017fff7ffd",
+    "0x6",
+    "0x10780017fff7fff",
+    "0x2",
+    "0x10780017fff7fff",
+    "0x7",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127ffd7fff8000",
+    "0x48127ffd7fff8000",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x208b7fff7fff7ffe",
+    "0xa0680017fff8000",
+    "0x6",
+    "0x40780017fff7fff",
+    "0x1",
+    "0x10780017fff7fff",
+    "0xb",
+    "0x40780017fff7fff",
+    "0x1",
+    "0x400080007fff7ffe",
+    "0x480680017fff8000",
+    "0x1",
+    "0x48127ffe7fff8000",
+    "0x482480017ffd8000",
+    "0x1",
+    "0x208b7fff7fff7ffe",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x480680017fff8000",
+    "0x0",
+    "0x208b7fff7fff7ffe",
+    "0x208b7fff7fff7ffe"
+  ],
+  "hints": [
+    [
+      138,
+      [
+        "\nmemory[ap + 0] = 0; \nroll(address=[fp + -4], caller_address=[fp + -3])\n"
+      ]
+    ],
+    [
+      144,
+      [
+        "memory[ap + 0] = segments.add()"
+      ]
+    ]
+  ],
+  "test_entry_points": [
+    {
+      "offset": 0,
+      "name": "examples::cheatcode_caller::test_cheatcode_caller"
+    },
+    {
+      "offset": 3,
+      "name": "examples::cheatcode_caller::test_cheatcode_caller_twice"
+    },
+    {
+      "offset": 6,
+      "name": "examples::cheatcode_caller::test_cheatcode_caller_three"
+    }
+  ]
```

Comment:
Why does this file contain "test_entry_points"?

Reply:
it is compiled test suite

Reply:
Wouldn't it better to compile to more abstract concepts?

```
"functions": [
    {
      "offset": 0,
      "name": "it_should_do_sth"
      "decorators": [{ name: "test" }]
    },
    {
      "offset": 3,
      "name": "it_should_do_sth_2"
      "decorators": [{ name: "test" }, {name: "skip", params: ["skip justification"]}]
    },
  ]
```

Reply:
Would be, but It is impossible to just run any function as an main in casm. Those are offsets to extra segments of casm which make it possible & call the function afterwards. Those segments would have to be generated for very function included in the sierra file 😢 .

Reply:
Does generating segments have significant impact on compilation performance? 

Resolving, I don't know design constraints.

Reply:
> Does generating segments have significant impact on compilation performance?
Probably not but I just don't see the point in generating them for all of the linked funcions